spads_config_bar_updater.py

```diff
@@ -62,7 +62,9 @@ def recursecopy(workdir):
 
 def configupdate(args):
 	if args.nogit != True:
-		execute("git pull https://github.com/beyond-all-reason/spads_config_bar.git")
+		execute("git fetch origin")
+		execute("git reset --hard FETCH_HEAD")
```

Comment:
`git fetch origin` will fetch *all* branches, everything from remote, (that can be changed by `remote.origin.fetch` option by by default it's just fetching everything. That means that `FETCH_HEAD` will contain per docs:

> The names of refs that are fetched, together with the object names they point at, are written to .git/FETCH_HEAD.

So, when I did, in the main bar game repo on my system:

```
git fetch upstream
git log FETCH_HEAD
```

I just got log of some random branch that was fetched, because `.git/FETCH_HEAD` actually contained 91 lines as that's how many references were updated.

In `rapid-hosting` we fetch a single named branch, so `FETCH_HEAD` contains exactly what we want.
 
---

I believe for this PR, `git pull --rebase` might be just good enough.

Alternative would be some crazy stuff like

```
BRANCH=$(git rev-parse --abbrev-ref HEAD)
git fetch origin $BRANCH
git checkout FETCH_HEAD --force
git clean -xfdf
git branch -f $BRANCH
git checkout $BRANCH
```

Reply:
`pull --rebase` won't create any commits, I don't recall it doing anything like that. Very good point about the script getting interrupted. The steps you propose look good to me, those should be fine.